app/code/Magento/InventoryCatalog/Model/ResourceModel/BulkInventoryTransfer.php

```diff
@@ -80,23 +80,23 @@ public function __construct(
     /**
      * @param string $sku
      * @param string $source
-     * @return float|null
+     * @return array|null
      */
-    private function getQuantityFromSource(string $sku, string $source): ?float
+    private function getSourceItem(string $sku, string $source): ?array
```

Comment:
it's better to call this function as `getSourceItemData` then getSourceItem, as we return the raw data from DB here